### DISCUSSION TITLE
Making graph output dependent on external variable

### DISCUSSION THREAD

**Discussion by kiddersmxj** [ArthurSonzogni/FTXUI#612] [Q&A] [answered]
Please forgive the seemingly rather simple question. However, I have read through all the Docs and spent many hours fiddling with example scripts and I cannot work this out. 

I am trying to pass a vector into the graph object to be displayed rather than just have an operation within the `operator()` that is dependent on the `shift` value to produce the graphed vector like the examples. How would I go about this? I am also reasonably new to cpp.

Thank you for you help.

**Comment by ArthurSonzogni** [comment, marked as answer]
I see the homescreen example is displaying a graph depending on the `shift` value:
https://github.com/ArthurSonzogni/FTXUI/blob/6215240b3722743b6e120b0734faaa8d538e0167/examples/component/homescreen.cpp#LL35C1-L46C5

**Reply by kiddersmxj** [reply]
Thanks so much! Sorted now :)
